test/bats/setup.bats

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bats
+
+BATS_LIBS_PATH="${LIBS_PATH:-../libs}" # fallback to relative path if not set
+load "${BATS_LIBS_PATH}/bats-assert/load"
+load "${BATS_LIBS_PATH}/bats-support/load"
+
+setup_file() {
+    uuid_v4_regex='^[0-9a-f]{8}-[0-9a-f]{4}-[4][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$'
+    ip_regex='^([0-9]{1,3}\.){3}[0-9]{1,3}(\/[0-9]{1,2})?$'
+}
+
+randStr() {
+    size=$1
+    cat /dev/urandom | tr -dc 'a-zA-Z0-9' | head -c "$size"
+}
+
+generate_ssh_key() {
+    ssh_key_path="id_rsa_$(randStr 8)"
+    ssh-keygen -t rsa -b 4096 -f "$ssh_key_path" -N "" >/dev/null 2>&1
+    echo $ssh_key_path
+}
+
+retry_command() {
+    local n=1
+    local max=3
+    local delay=10
+    while true; do
+        "$@" && break || {
+            if [[ $n -lt $max ]]; then
+                ((n++))
+                echo "Command failed. Attempt $n/$max:"
+                sleep $delay;
+            else
+                fail "$n attempts exhausted for command: $*"
+                break
+            fi
+        }
+    done
+}
+
+find_or_create_resource() {
```

Comment:
Is there a way in which we could make this function accept resources that have plain numeric ids as well? I think it would be useful for testing scenarios which require, for example, preexisting LANs.

Reply:
Yeah we could probably just get rid of that `assert_regex` statement, and the callee should check whatever it wants to check